src/System.Management.Automation/utils/StringUtil.cs

```diff
@@ -81,6 +81,24 @@ internal static string Padding(int countOfSpaces)
 
             return result;
         }
+
+        private const int DashCacheMax = 120;
+        private static readonly string[] DashCache = new string[DashCacheMax];
```

Comment:
Seems like we can leverage the new Span<T> and Slice capabilities of C# 7.2

Reply:
Good catch! I like the idea.

Reply:
TableWriter expect strings, pass them through many methods and collect them in StringCollection.
So I can not implement the idea - it requires a complete rewriting of this TableWriter and it's a big risk - seems we haven't full set tests.
I am ready to pull another PR with some optimizations in TableWriter to decrease allocations.

I suggest use the cache until we're ready to rewrite TableWriter using `Memory<char>` or something else.